scripts/sasl-xoauth2-tool.in

```diff
@@ -219,6 +227,14 @@ def argparse_get_parser() -> argparse.ArgumentParser:
 
 
 def subcommand_get_token(args:argparse.Namespace) -> None:
+    input_dict = {}
+    if args.overwrite_existing_token:
+        if not os.path.isfile(args.output_file):
+            raise Exception("Cannot overwrite nonexistent token file {}".format(args.output_file))
+        else:
+            with open(args.output_file,'r') as input_file:
+                input_dict = json.load(input_file)
+    args.output_file = open(args.output_file,'w')
```

Comment:
So in principle I agree that it's better for us to pass a file object around (rather than a path), but in this case it has the unfortunate effect of leaving a zero-length file sitting around if fetching the token fails. AFAICT this one line isn't strictly required for the rest of your patch to work, right?

Reply:
No, I was just trying to keep it as similar as possible to the previous implementation, in which the file object was immediately created by argparse. But I agree that it's better to not to open the file for writing until we're sure the token call succeeded, since otherwise it clears out the file contents, which does not preserve the extra fields.

Reply:
Oh I totally missed that in the initial implementation. Thanks for changing it!